### PR TITLE
Move ringpop code from membership to ringpop package

### PR DESCRIPTION
**Pull request by MichaelSnowden**
**What changed?**
I moved all ring RingPop-related code from the membership package to the ringpop package.
I also addressed several linter warnings.

**Why?**
To separate interfaces from implementations.

**How did you test it?**
Existing tests.

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: common/ringpop/monitor_test.go [hunk 1]
```diff
@@ -22,14 +22,15 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package membership
+package ringpop
```
Comment (yycptt) [inline common/ringpop/monitor_test.go:25]: nit: L53 `TestRingpopMonitor` -> `TestMonitor`

File: common/ringpop/ringpop_factory.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
+// The MIT License
```
Comment (yycptt) [inline common/ringpop/ringpop_factory.go:1]: nit: `ringpop_factory.go` -> `factory.go`?

File: common/ringpop/ringpop_factory.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+	defaultMaxJoinDuration      = 10 * time.Second
+	persistenceOperationTimeout = 10 * time.Second
+)
+
+var (
+	IPV4Localhost = net.IPv4(127, 0, 0, 1)
+)
+
+// RingpopFactory implements the RingpopFactory interface
+type ringpopFactory struct {
```
Comment (yycptt) [inline common/ringpop/ringpop_factory.go:60]: nit: now that the package is `ringpop`, `ringpopFactory` -> `factory`? same for `newRingpopFactory` and `ringpopServiceResolver`

File: common/ringpop/ringpop_provider.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
+// The MIT License
```
Comment (yycptt) [inline common/ringpop/ringpop_provider.go:1]: nit: `fx.go` or `provider.go`?

**Review by yycptt — APPROVED** [review]
Does it make sense to make ringpop a sub-package of membership?

**Comment by yycptt** [comment]
There's something wrong with the rebase? looks like other changes got included into this PR.

**Comment by MichaelSnowden** [comment]
> There's something wrong with the rebase? looks like other changes got included into this PR.

Thanks for catching that. Yeah, it was a bad rebase. Fixed now

**Pull request merged**